src/libasr/codegen/x86_assembler.h

```diff
@@ -567,6 +641,21 @@ class X86Assembler {
         EMIT("mov " + r2s(r32) + ", " + i2s(imm32));
     }
 
+    void rex(int W, int R, int X, int B) {
+        LFORTRAN_ASSERT(W <= 1);
+        LFORTRAN_ASSERT(R <= 1);
+        LFORTRAN_ASSERT(X <= 1);
+        LFORTRAN_ASSERT(B <= 1);
+        m_code.push_back(m_al, (0b01000000 | (W << 3) | (R << 2) | (X << 1) | B));
+    }
```

Comment:
`W`, `R`, `X`, `B` are single bit values (they take value `0` or `1`), and I am unsure what `data type` (`int`, `uint8_t`, etc.) they should have. I currently used `int` data type for them (as `int` seems general).

Reply:
I would use the same data type that m_code.push_back() is using, which is `uint8_t`, so I would use that. That way there are no issues with automatic casting.

Reply:
Got it, Thank you.